### PR TITLE
Fixes #20971: Adapt the DSC code generator to the new ncf syntax

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/20971

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

File: language/src/generator/dsc/syntax.rs [hunk 1]
```diff
@@ -487,17 +555,77 @@ impl Call {
...
+            )
         )
     }
 
+    pub fn map(name: &str, content: &HashMap<String,String>, extra_ref: Option<String>) -> Self {
+        let mut map_as_string = match extra_ref {
+            None => "@{\n".to_string(),
+            Some(i) => format!(
+                "{} + @{{\n",
+                Some(i).unwrap()
```
Comment (peckpeck) [inline language/src/generator/dsc/syntax.rs:567]: this is called "i"

**Comment by fanf** [comment]
Closing this one since se changed the generation path for 7.2 and rudderc is not involved anymore for now

**Pull request closed without merging**